ProcessLib/HeatTransportBHE/BHE/BHE_1P.h

```diff
@@ -0,0 +1,158 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <Eigen/Eigen>
+
+#include "BaseLib/Error.h"
+
+#include "BHECommon.h"
+#include "FlowAndTemperatureControl.h"
+#include "PipeConfiguration1PType.h"
+
+namespace ProcessLib
+{
+namespace HeatTransportBHE
+{
+namespace BHE
+{
+/**
+ * The BHE_1P class is the realization of single-pipe type of Borehole Heate
+ * Exchanger. In this class, the pipe heat capacity, pipe heat conductiion, pie
+ * advection vectors are intialized according to the geometry of the single-pipe
+ * type of BHE. For this type of BHE, 2 primary unknowns are assigned on the 1D
+ * BHE elements. They are the temperature in the pipe T_p, and temperature of
+ * the grout zone sorrounding the single pipe T_g. These two primary varaibles
+ * are solved according to heat convection and conduction equations on the pipes
+ * and also in the grout zones. The interaction of the 1P type of BHE and the
+ * sorrounding soil is regulated through the thermal resistance values, which
+ * are calculated specifically during the initialization of the class.
+ */
+class BHE_1P final : public BHECommon
+{
+public:
+    BHE_1P(BoreholeGeometry const& borehole,
```

Comment:
Most, if not all of the implementations could go into the cpp.

Reply:
Moved the constructor and calculation of cross section areas into cpp file.